src/arrayMethodJoin.js

```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let divider = separator;
+
+    if (divider === undefined) {
+      divider = ',';
```

Comment:
This condition is redundant, as the default value for 'separator' has already been set in the function definition. You can remove lines 11-15.